src/ui/page/dataverse/dataspace/governance/governance.tsx

```diff
@@ -0,0 +1,90 @@
+import type { FC } from 'react'
+import { useState, useCallback, useEffect } from 'react'
+import { useParams, useNavigate } from 'react-router-dom'
+import { pipe } from 'fp-ts/lib/function'
+import type { Option } from 'fp-ts/Option'
+import * as O from 'fp-ts/Option'
+import type { DataSpace } from '@/ui/page/dataverse/dataverse'
+import { getResourceDetails } from '@/ui/page/dataverse/dataverse'
+import { isDataSpace } from '@/ui/page/dataverse/dataspace/dataspace'
+import { GovernanceContent } from './governanceContent'
+import { mockedGovernanceChapter } from './mockedData'
+import './governance.scss'
+
+// eslint-disable-next-line max-lines-per-function
+export const Governance: FC = () => {
+  const navigate = useNavigate()
+  const {
+    id: dataspaceId,
+    sectionId: sectionIdParams,
+    subsectionId: subsectionIdParams
+  } = useParams<string>()
+
+  const [dataspace, setDataspace] = useState<Option<DataSpace>>(O.none)
+
+  const sections = mockedGovernanceChapter.contains
+  const currentSection = sections.find(section => section.id === sectionIdParams)
```

Comment:
I think there is a failure in the component conception, because you are looking for values based upon optional params..
IMHO, you should:

- test params availability and handle rendering accordingly 
- then memoize computations based upon params
- then construct `uri` for the redirection